improver/wxcode/utilities.py

```diff
@@ -243,3 +266,178 @@ def get_parameter_names(diagnostic_fields: List[List[str]]) -> List[List[str]]:
         elif is_variable(condition):
             parameter_names.append(condition)
     return parameter_names
+
+
+def _check_diagnostic_lists_consistency(query: Dict[str, Any]) -> bool:
+    """
+    Checks if specific input lists have same nested list
+    structure. e.g. ['item'] != [['item']]
+
+    Args:
+        query (dict):
+            of weather-symbols decision-making information
+
+    Raises:
+        ValueError: if diagnostic query lists have different nested list
+            structure.
+
+    """
+    diagnostic_keys = [
+        "diagnostic_fields",
+        "diagnostic_conditions",
+        "diagnostic_thresholds",
+    ]
+    values = [
+        get_parameter_names(query[key]) if key == "diagnostic_fields" else query[key]
+        for key in diagnostic_keys
+    ]
+    return _check_nested_list_consistency(values)
+
+
+def _check_nested_list_consistency(query: List[List[Any]]) -> bool:
+    """
+    Return True if all input lists have same nested list
+    structure. e.g. ['item'] != [['item']]
+
+    Args:
+        query (list of lists):
+            Nested lists to check for consistency.
+
+    Returns:
+        bool: True if diagnostic query lists have same nested list
+            structure.
+
+    """
+
+    def _checker(lists):
+        """Return True if all input lists have same nested list
+        structure. e.g. ['item'] != [['item']]."""
+        type_set = set(map(type, lists))
+        if list in type_set:
+            return (
+                len(type_set) == 1
+                and len(set(map(len, lists))) == 1
+                and all(map(_checker, zip(*lists)))
+            )
+        return True
+
+    return _checker(query)
+
+
+def check_tree(wxtree: Dict[str, Dict[str, Any]]) -> str:
```

Comment:
Reinterprets old static unit tests to provide checking functionality for decision trees:
https://github.com/metoppv/improver/blob/master/improver_tests/wxcode/wxcode/test_wxcode_decision_tree.py